File: sokoban-server/src/main/java/com/dqv5/sokoban/common/BlockType.java
package com.dqv5.sokoban.common;

import java.util.Arrays;

public enum BlockType {
    MAN("@"),
    MAN_ON_GOAL("+"),
    BOX("$"),
    BOX_ON_GOAL("*"),
    WALL("#"),
    GOAL("."),
    FLOOR("-");

    private final String text;

    BlockType(String text) {
        this.text = text;
    }

    public static BlockType fromText(String aChar) {
        return Arrays.stream(values()).filter(x -> x.text.equals(aChar)).findFirst().orElseThrow(() -> new RuntimeException("参数错误"));
    }


    public String getText() {
        return text;
    }
}

File: solver/blockType.js
class BlockType {
    static MAN = '@';
    static MAN_ON_GOAL = '+';
    static BOX = '$';
    static BOX_ON_GOAL = '*';
    static WALL = '#';
    static GOAL = '.';
    static FLOOR = '-';

    static ALL_LIST = [
        this.MAN,
        this.MAN_ON_GOAL,
        this.BOX,
        this.BOX_ON_GOAL,
        this.WALL,
        this.GOAL,
        this.FLOOR
    ]


    static parse(text) {
        return this.ALL_LIST.find(x => x === text)
    }

}

exports.BlockType = BlockType
